### PR TITLE
New provider: OpenStack

### PR DESCRIPTION
**Pull request by dnsmichi**
Thanks @martbhell for the initial implementation.

### CODE REVIEW

**Comment by martbhell** [comment]
Cool stuff!
I guess this is still a work in progress? Would be nice with some documentation how to use this.

I needed to set:
 - OS_SSH_USER
 - OS_SSH_PRIVATE_KEY_PATH
 - OS_KEYPAIR_NAME
 - OS_IMAGE
 - OS_FLAVOR

and also modify the Vagrantfile (or create a security group called Vagrant)

I still got the "Unsupported OS" on my CentOS-7 image - because https://github.com/Icinga/icinga-vagrant/blob/master/.puppet/scripts/shell_provisioner_pre.sh#L19 conditional was not cool - made a PR for that: #171>
After that the vagrant and puppet completed and I could login to an icinga web interface :)

File: standalone/Vagrantfile [hunk 1]
```diff
@@ -61,6 +62,30 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
...
+      node_config.vm.provider :openstack do |os, override|
+        override.ssh.username  = "#{ENV['OS_SSH_USER']}"
+        override.ssh.private_key_path = "#{ENV['OS_SSH_PRIVATE_KEY_PATH']}"
+        os.meta_args_support      = true
+        os.openstack_auth_url     = "#{ENV['OS_AUTH_URL']}"
+        os.username               = "#{ENV['OS_USERNAME']}"
+        os.password               = "#{ENV['OS_PASSWORD']}"
+        os.flavor                 = "#{ENV['OS_FLAVOR']}" # TODO: config
+        os.image                  = "#{ENV['OS_IMAGE']}" # TODO: config
+        os.project_name           = "#{ENV['OS_PROJECT_NAME']}" # TODO: config
```
Comment (martbhell) [inline standalone/Vagrantfile]: OS_PROJECT_NAME should not be a config/todo either, it comes from the openrc file too.

**Comment by dnsmichi** [comment]
Docs was still in my local working copy, it is now pushed.

**Comment by dnsmichi** [comment]
I've also refactored the Vagrantfile files for each project into generic helper functions, which allows for easier modifications.

**Comment by dnsmichi** [comment]
The config TODOs have been removed, this is left for the user to set.

**Comment by dnsmichi** [comment]
I've tested this locally with Parallels, remotely with Openstack for standalone, influxdb and elastic. It is important to open up the security groups and exposed ports to allow intercommunication via external floating ip. This affects Graphite, InfluxDB, Elastic, etc. and is especially noted in the documentation.

The docs are moved to the `doc/` btw, as it doesn't fit into the general README (which is for basic users, not advanced OpenStack users).

**Comment by dnsmichi** [comment]
Need to fix a parameter problem for Elasticsearch http proxy, fix is coming.

**Comment by dnsmichi** [comment]
Fixed it, needed to move the mapped port range for Kibana/ES to 5602/9202.

**Pull request merged**